src/library/scala/collection/Iterable.scala

```diff
@@ -549,21 +549,21 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
     *
     */
   def groupBy[K](f: A => K): immutable.Map[K, C] = {
-    val m = mutable.Map.empty[K, Builder[A, C]]
+    val m = immutable.HashMap.newBuilder[K, Builder[A, C]].asInstanceOf[immutable.HashMapBuilder[K, Builder[A, C]]]
     val it = iterator
     while (it.hasNext) {
       val elem = it.next()
       val key = f(elem)
-      val bldr = m.getOrElseUpdate(key, newSpecificBuilder)
+      val bldr = m.getOrElse(key, null) match {
```

Comment:
What if the collection is a collection of `null` values?

Reply:
This is the collection we're building, so we know values can only be whatever  `newSpecificBuilder` returns, which we're assuming to be non-null.